From 67ea7fe899f52b7c82528dd7be57dd42e22f296b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 28 Oct 2024 13:53:39 +0100
Subject: [PATCH] Make reveal.js optional

If reveal.js is not present on the system, don't set the `revealjs-url`
variable, so it will be loaded from unpkg.com.
---
 apostrophe/export_dialog.py | 7 ++++---
 meson.build                 | 4 ----
 2 files changed, 4 insertions(+), 7 deletions(-)

diff --git a/apostrophe/export_dialog.py b/apostrophe/export_dialog.py
index 0fd93be0..bf723bcb 100644
--- a/apostrophe/export_dialog.py
+++ b/apostrophe/export_dialog.py
@@ -209,6 +209,7 @@ class AdvancedExportDialog(Adw.Dialog):
 
         self.file = file
         self.text = text
+        self.revealjs = helpers.get_media_path('/libs/reveal.js')
 
         if not self.formats:
             with open(helpers.get_media_path("/media/formats.json")) as f:
@@ -303,8 +304,8 @@ class AdvancedExportDialog(Adw.Dialog):
         def on_response(dialog, response):
             if self.exports_multiple_files:
                 folder = export_dialog.get_file()
-                shutil.copytree(helpers.get_media_path('/libs/reveal.js'),
-                                os.path.join(folder.get_path(), 'reveal.js'), dirs_exist_ok=True)
+                if self.revealjs:
+                    shutil.copytree(self.revealjs, os.path.join(folder.get_path(), 'reveal.js'), dirs_exist_ok=True)
                 export_file = folder.get_child(self.file.name + '.' +
                                 self.formats_list.get_selected_row().item.ext)
             else:
@@ -387,7 +388,7 @@ class AdvancedExportDialog(Adw.Dialog):
             if self.sw_incremental_bullets.get_active():
                 args.append("--incremental")
 
-        if self.formats_list.get_selected_row().item.to == "revealjs":
+        if self.formats_list.get_selected_row().item.to == "revealjs" and self.revealjs:
             args.extend(["-V", "revealjs-url=reveal.js"])
 
         return args
diff --git a/meson.build b/meson.build
index 6dd4ca89..8b2d1a1e 100644
--- a/meson.build
+++ b/meson.build
@@ -60,10 +60,6 @@ pkgdatadir = datadir / meson.project_name()
 #bindir = join_paths(get_option('prefix'), get_option('bindir'))
 podir = meson.source_root() / 'po'
 
-if not fs.exists(join_paths(pkgdatadir, 'libs/reveal.js'))
-  error('No valid reveal.js installation found')
-endif
-
 subdir('data')
 #subdir('help')
 subdir('po')
-- 
GitLab

